streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1343,6 +1348,17 @@ private long getCacheSizePerThread(final int numStreamThreads) {
         return totalCacheSize / (numStreamThreads + (topologyMetadata.hasGlobalTopology() ? 1 : 0));
     }
 
+    private long getMaxUncommittedBytesPerThread(final int numStreamThreads) {
```

Comment:
> Does passing maxUncommittedBytesPerThread already affect commit behavior? As far as I understand the code it does not because approximateNumUncommittedBytes() returns always 0 at the moment.

No, for exactly this reason.

> This method does not consider whether default.state.isolation.level is set to READ_COMMITTED.

Does it need to? Under READ_UNCOMMITTED, since transaction buffers aren't used, `approximateNumUncommittedBytes()` will always return `0` (as it does right now), ensuring that we never exceed the configured threshold.

Notably, if we make this dependent on the isolation level, any future changes that add support for transaction buffers to READ_UNCOMMITTED writes would break this.